lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -414,7 +414,8 @@ def set_server_facts
         @server_facts["servername"] = host
       end
     end
-    if @server_facts["serverip"].nil?
+
+    if @server_facts["serverip"].nil? && @server_facts["serverip6"].nil?
       Puppet.warning _("Could not retrieve fact serverip")
```

Comment:
nit, maybe update message that server facts for serverip and serverip6 could not be resolved?